plugins/modules/net_tools/gandi_livedns.py

```diff
@@ -0,0 +1,215 @@
+- name: Create a test A record to point to 127.0.0.1 in the my.com zone.
+  gandi_livedns:
+    zone: my.com
+    record: test
+    type: A
+    values:
+    - 127.0.0.1
+    api_key: dummyapitoken
+  register: record
+
+- name: Create a mail CNAME record to www.my.com domain
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    record: mail
+    values:
+    - www
+    api_key: dummyapitoken
+    state: present
+
+- name: Change its TTL
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    values:
+    - www
+    ttl: 10800
+    api_key: dummyapitoken
+    state: present
+
+- name: Delete the record
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    record: mail
+    api_key: dummyapitoken
+    state: absent
+'''
+
+RETURN = r'''
+record:
+    description: A dictionary containing the record data.
+    returned: success, except on record deletion
+    type: complex
+    contains:
+        values:
+            description: The record content (details depend on record type).
+            returned: success
+            type: list
+            sample:
+            - 192.0.2.91
+            - 192.0.2.92
+        record:
+            description: The record name.
+            returned: success
+            type: str
+            sample: www
+        ttl:
+            description: The time-to-live for the record.
+            returned: success
+            type: int
+            sample: 300
+        type:
+            description: The record type.
+            returned: success
+            type: str
+            sample: A
+        domain:
+            description: The domain associated with the record.
+            returned: success
+            type: str
+            sample: my.com
+        zone:
+            description: The zone associated with the record.
+            returned: success
+            type: str
+            sample: my.com
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.gandi_livedns_api import GandiLiveDNSAPI
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            api_key=dict(type='str', required=True, no_log=True),
+            record=dict(type='str', default='@', aliases=['name']),
+            state=dict(type='str', default='present', choices=['absent', 'present']),
+            ttl=dict(type='int', default=None),
+            type=dict(type='str', required=True,
+                      choices=['A', 'AAAA', 'ALIAS', 'CAA', 'CDS', 'CNAME', 'DNAME', 'DS', 'KEY', 'LOC', 'MX', 'NS', 'PTR', 'SPF', 'SRV',
+                               'SSHFP', 'TLSA', 'TXT', 'WKS']),
+            values=dict(type='list', aliases=['content'], elements='str'),
+            zone=dict(type='str'),
+            domain=dict(type='str'),
+        ),
+        supports_check_mode=True,
+        required_if=[
+            ('state', 'present', ['record', 'type', 'values']),
+            ('state', 'absent', ['record', 'type']),
```

Comment:
Ssince "type" is already listed as required, shouldn't it be removed from here ? 
And record could also be listed as required, which in turn would simplify, since everything but values would be required=True.

Reply:
I'll remove "type" but "record" has a default ("@") so it cannot be required=True